File: backend/src/main/java/com/revnu/backend/features/analytics/dto/DailyAnalyticsResponse.java
package com.revnu.backend.features.analytics.dto;

import java.math.BigDecimal;
import java.util.List;

public record DailyAnalyticsResponse(
        BigDecimal totalSales,
        BigDecimal totalExpenses,
        BigDecimal totalSalaries,
        BigDecimal netProfit,
        BigDecimal yesterdaySales,
        BigDecimal yesterdayExpenses,
        BigDecimal yesterdayProfit,
        long saleRecordsCount,
        long expenseRecordsCount,
        List<SalesTrendPoint> salesTrend,
        List<TagSlice> categories,
        String yesterdayLabel,
        String date,
        boolean isClosed
        ) {

}

File: mobile/app/src/main/java/com/revnu/mobile/features/analytics/model/DailyAnalyticsResponse.kt
package com.revnu.mobile.features.analytics.model

data class DailyAnalyticsResponse(
    val totalSales: Double,
    val totalExpenses: Double,
    val totalSalaries: Double,
    val netProfit: Double,
    val yesterdaySales: Double,
    val yesterdayExpenses: Double,
    val yesterdayProfit: Double,
    val saleRecordsCount: Long,
    val expenseRecordsCount: Long,
    val isClosed: Boolean,
    val date: String
)